app/models/concerns/hostext/search.rb

```diff
@@ -191,6 +191,22 @@ def param_conditions(p)
         end
         conditions.empty? ? [] : "( #{conditions.join(' OR ')} )"
       end
+
+      def cast_facts(key, operator, value)
+        logger.info("Key: #{key} Operator: #{operator} Value: #{value}")
+        begin
+          is_int = !!Integer(value)
+        rescue ArgumentError, TypeError
+          is_int = false
+        end
+        if is_int
+          opts = "fact_names.name = '#{key.split('.')[1]}' AND CAST(fact_values.value AS INTEGER) #{operator} #{value}"
+        else
+          opts = "fact_names.name = '#{key.split('.')[1]}' AND fact_values.value #{operator} #{value}"
+        end
+        logger.info("Condition: #{opts}")
```

Comment:
Yeah, as said - this is not the final code. @ohadlevy said we should discuss the best way to implement it over a PR, so this is just to kickoff a discussion :)